lib/block-supports/elements.php

```diff
@@ -65,11 +65,21 @@ function gutenberg_render_elements_support( $block_content, $block ) {
 		$content              = substr_replace( $block_content, ' class="' . $class_name . '"', $first_element_offset + strlen( $first_element ) - 1, 0 );
 	}
 
-	// Ideally styles should be loaded in the head, but blocks may be parsed
-	// after that, so loading in the footer for now.
+	// Styles should be loaded in the head for block themes
+	// and classic themes by default.
+	//
+	// Only for classic themes that opt into loading separate block assets
+	// we should load the styles in the body.
+	//
 	// See https://core.trac.wordpress.org/ticket/53494.
+	$separate_assets  = wp_should_load_separate_core_block_assets();
+	$is_classic_theme = ! wp_is_block_theme();
+	$action_hook_name = 'wp_enqueue_scripts';
+	if ( $is_classic_theme && $separate_assets ) {
+		$action_hook_name = 'wp_footer';
+	}
 	add_action(
-		'wp_footer',
+		$action_hook_name,
 		function () use ( $style ) {
 			echo $style;
 		}
```

Comment:
Instead of repeating the same logic over and over between block supports, can we have a function like `wp_enqueue_block_style( style )` or similar and reuse it across block supports?

Reply:
https://github.com/WordPress/gutenberg/pull/38750/commits/8cbc00ecf49d632428d01003edb96f344e3526c0